src/dc_receive_imf.rs

```diff
@@ -521,27 +522,36 @@ async fn add_parts(
 
         if chat_id.is_unset() {
             // check if the message belongs to a mailing list
-            if let Some(list_id_header) = mime_parser.get(HeaderDef::ListId) {
-                let create_blocked = Blocked::Deaddrop;
-
-                let (new_chat_id, new_chat_id_blocked) = create_or_lookup_mailinglist(
-                    context,
-                    allow_creation,
-                    create_blocked,
-                    list_id_header,
-                    &mime_parser.get_subject().unwrap_or_default(),
-                )
-                .await;
-
-                *chat_id = new_chat_id;
-                chat_id_blocked = new_chat_id_blocked;
-                if let Some(from) = mime_parser.from.first() {
-                    if let Some(from_name) = &from.display_name {
-                        for part in mime_parser.parts.iter_mut() {
-                            part.param.set(Param::OverrideSenderDisplayname, from_name);
-                        }
+            match mime_parser.get_mailinglist_type() {
+                MailinglistType::ListIdBased => {
+                    if let Some(list_id) = mime_parser.get(HeaderDef::ListId) {
+                        let (new_chat_id, new_chat_id_blocked) = create_or_lookup_mailinglist(
+                            context,
+                            allow_creation,
+                            Blocked::Deaddrop,
+                            list_id,
+                            &mime_parser.get_subject().unwrap_or_default(),
+                        )
+                        .await;
+                        *chat_id = new_chat_id;
+                        chat_id_blocked = new_chat_id_blocked;
                     }
                 }
+                MailinglistType::SenderBased => {
+                    if let Some(sender) = mime_parser.get(HeaderDef::Sender) {
+                        let (new_chat_id, new_chat_id_blocked) = create_or_lookup_mailinglist(
+                            context,
+                            allow_creation,
+                            Blocked::Deaddrop,
+                            sender,
+                            &mime_parser.get_subject().unwrap_or_default(),
+                        )
+                        .await;
+                        *chat_id = new_chat_id;
+                        chat_id_blocked = new_chat_id_blocked;
```

Comment:
Just wondering if this could be done without the 10 lines of duplicate code (these 10 lines are also above, just with `list_id` instead of `sender`). It could be done by a function in `mimeparser.rs` that replaces `get_mailinglist_type()` and directly returns the `list_id_header` parameter for `create_or_lookup_mailinglist()`. I can't come up with a good name for this function though.

It's not overly important, 10 lines of duplicate code aren't the worst thing :)

Reply:
i think, these duplicated lines are okay here. it is a bit blown up by formatting, without that, it is a function call with different arguments and then saving the returned value to locals.

Reply:
but i just noticed, the `Blocked::Deaddrop` argument is superfluous, will do a commit.